microbench-ci: refactor posting

herkolategan opened the pull request:
This change ensures that for post `microbench-ci` is built from base, which is more secure given this part of the job has access to PRs & posting on GH. The change also adds the pull requests write permissions required for posting comments to a PR.

Epic: None
Release note: None

herkolategan committed:
microbench-ci: post should build from base

For security reasons, it's better if the post section of the workflow builds
`microbench-ci` from base.

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/140976)
<!-- Reviewable:end -->

herkolategan committed:
microbench-ci: add workflow permission

The post part for the Microbenchmark CI workflow seems to require the
`pull-requests` permission in addition to the `issues` permission to be able to
post results to a PR.

Epic: None
Release note: None

herkolategan commented:
TFTR!

bors r=rickystewart
